FAILED: akka.remote.ActorsLeakSpec

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/7033/consoleFull

```
[info] - must not leak actors *** FAILED *** (24 seconds, 618 milliseconds)
[info]   Expected Set(Actor[akka://ActorsLeakSpec/system/endpointManager#1423904439], Actor[akka://ActorsLeakSpec/system/transports#1152699934], Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp61#-1527258615], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp14#1344748334]), but got Set(Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp61#-1527258615], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp14#1344748334], Actor[akka://ActorsLeakSpec/system/transports#1152699934], Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A50508-9#-245584626], Actor[akka://ActorsLeakSpec/system/endpointManager#1423904439]) (ActorsLeakSpec.scala:209)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions$class.newAssertionFailedException(Assertions.scala:495)
[info]   at akka.testkit.AkkaSpec.newAssertionFailedException(AkkaSpec.scala:57)
[info]   at org.scalatest.Assertions$class.assertResult(Assertions.scala:1226)
[info]   at akka.testkit.AkkaSpec.assertResult(AkkaSpec.scala:57)
[info]   at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(ActorsLeakSpec.scala:209)
[info]   at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ActorsLeakSpec.scala:69)
[info]   at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ActorsLeakSpec.scala:69)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:57)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:950)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:962)
[info]   at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:57)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1021)
[info]   at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:57)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1424)
[info]   at akka.testkit.AkkaSpec.org$scalatest$WordSpecLike$$super$run(AkkaSpec.scala:57)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1067)
[info]   at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:57)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:257)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:256)
[info]   at akka.testkit.AkkaSpec.run(AkkaSpec.scala:57)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
[info]   at sbt.TestRunner.runTest$1(TestFramework.scala:76)
[info]   at sbt.TestRunner.run(TestFramework.scala:85)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
```

Out of 10 times of running this testcase, it failed once on my machine. Logs:
```
[INFO] [08/08/2016 16:19:25.380] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.384] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:43873]
[INFO] [08/08/2016 16:19:25.384] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:43873]
[INFO] [08/08/2016 16:19:25.499] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43873/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.501] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43873/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.521] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A43873-0/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:43873]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:43873] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:43873
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.526] [remote-akka.actor.default-dispatcher-4] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.528] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43873/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.553] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.556] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:37885]
[INFO] [08/08/2016 16:19:25.557] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:37885]
[INFO] [08/08/2016 16:19:25.575] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:37885/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.576] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:37885/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.577] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A37885-1/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:37885]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:37885] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:37885
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.581] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.582] [remote-akka.remote.default-remote-dispatcher-8] [akka.trttl.tcp://remote@127.0.0.1:37885/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.598] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.600] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:34599]
[INFO] [08/08/2016 16:19:25.601] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:34599]
[INFO] [08/08/2016 16:19:25.618] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:34599/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.618] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:34599/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.620] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A34599-2/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:34599]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:34599] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:34599
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.621] [remote-akka.actor.default-dispatcher-3] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.622] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:34599/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.630] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.633] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.633] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.656] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.656] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.658] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-3/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.661] [remote-akka.actor.default-dispatcher-4] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.661] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.675] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.679] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.679] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.700] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.700] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.701] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-4/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.702] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.702] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.715] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.718] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.719] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/08/2016 16:19:25.739] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.739] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/08/2016 16:19:25.741] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-5/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/08/2016 16:19:25.744] [remote-akka.actor.default-dispatcher-3] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.744] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:25.764] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:25.768] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:39695]
[INFO] [08/08/2016 16:19:25.768] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:39695]
[INFO] [08/08/2016 16:19:25.788] [remote-akka.remote.default-remote-dispatcher-8] [akka.trttl.tcp://remote@127.0.0.1:39695/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:25.788] [remote-akka.remote.default-remote-dispatcher-8] [akka.trttl.tcp://remote@127.0.0.1:39695/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/08/2016 16:19:25.791] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:25.791] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:39695/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:29.812] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:29.815] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:38027]
[INFO] [08/08/2016 16:19:29.815] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:38027]
[INFO] [08/08/2016 16:19:29.837] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38027/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:29.837] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38027/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/08/2016 16:19:29.840] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:29.840] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:38027/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:33.856] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:33.863] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:43675]
[INFO] [08/08/2016 16:19:33.863] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:43675]
[INFO] [08/08/2016 16:19:33.884] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43675/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:33.884] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43675/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/08/2016 16:19:33.885] [remote-akka.actor.default-dispatcher-3] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:33.886] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:43675/system/remoting-terminator] Remoting shut down.
[INFO] [08/08/2016 16:19:37.904] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/08/2016 16:19:37.908] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:36321]
[INFO] [08/08/2016 16:19:37.908] [pool-43-thread-2-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:36321]
[INFO] [08/08/2016 16:19:37.943] [remote-akka.remote.default-remote-dispatcher-9] [akka.trttl.tcp://remote@127.0.0.1:36321/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/08/2016 16:19:37.943] [remote-akka.remote.default-remote-dispatcher-9] [akka.trttl.tcp://remote@127.0.0.1:36321/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/08/2016 16:19:39.962] [remote-akka.actor.default-dispatcher-7] [akka://remote/deadLetters] Message [akka.remote.EndpointWriter$FlushAndStop$] from Actor[akka://remote/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2FActorsLeakSpec%40127.0.0.1%3A2552-0/endpointWriter#-1063763136] to Actor[akka://remote/deadLetters] was not delivered. [1] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[WARN] [08/08/2016 16:19:39.963] [remote-akka.actor.default-dispatcher-4] [akka.remote.Remoting] Shutdown finished, but flushing might not have been successful and some messages might have been dropped. Increase akka.remote.flush-wait-on-shutdown to a larger value to avoid this.
[INFO] [08/08/2016 16:19:39.963] [remote-akka.actor.default-dispatcher-4] [akka.remote.Remoting] Remoting shut down
[INFO] [08/08/2016 16:19:39.963] [remote-akka.remote.default-remote-dispatcher-9] [akka.trttl.tcp://remote@127.0.0.1:36321/system/remoting-terminator] Remoting shut down.
[WARN] [08/08/2016 16:19:49.982] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-5] [akka.remote.Remoting] Association to [akka.trttl.tcp://remote@127.0.0.1:36321] having UID [584617209] is irrecoverably failed. UID is now quarantined and all messages to this UID will be delivered to dead letters. Remote actorsystem must be restarted to recover from this situation.
```

There was also an instance where assertion failure was printed out, but test case passed:
```
akka > akka-remote/testOnly *.ActorsLeakSpec
[info] ActorsLeakSpec:
Coroner Thread Count starts at 22 in akka.remote.ActorsLeakSpec
[info] Remoting
[INFO] [08/09/2016 09:25:48.823] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:48.826] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:34987]
[INFO] [08/09/2016 09:25:48.826] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:34987]
[INFO] [08/09/2016 09:25:48.916] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:34987/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:48.917] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:34987/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:48.931] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A34987-0/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:34987]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:34987] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:34987
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:48.933] [remote-akka.actor.default-dispatcher-4] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:48.934] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:34987/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:48.957] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:48.960] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:34301]
[INFO] [08/09/2016 09:25:48.960] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:34301]
[INFO] [08/09/2016 09:25:48.978] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:34301/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:48.978] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:34301/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:48.980] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A34301-1/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:34301]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:34301] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:34301
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:48.981] [remote-akka.actor.default-dispatcher-5] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:48.981] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:34301/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:48.988] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:48.990] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:40193]
[INFO] [08/09/2016 09:25:48.991] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:40193]
[INFO] [08/09/2016 09:25:49.007] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:40193/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:49.007] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:40193/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:49.009] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A40193-2/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:40193]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:40193] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:40193
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:49.011] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:49.011] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:40193/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:49.021] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:49.024] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.025] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.046] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:49.046] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:49.047] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-3/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:49.049] [remote-akka.actor.default-dispatcher-3] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:49.049] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:49.063] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:49.066] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.066] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.085] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:49.085] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:49.086] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-4/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:49.088] [remote-akka.actor.default-dispatcher-3] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:49.088] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:49.097] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:49.100] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.100] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:2553]
[INFO] [08/09/2016 09:25:49.118] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:49.118] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [08/09/2016 09:25:49.119] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.0.1%3A2553-5/endpointWriter] AssociationError [akka.trttl.tcp://ActorsLeakSpec@127.0.0.1:2552] <- [akka.trttl.tcp://remote@127.0.0.1:2553]: Error [Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553] [
akka.remote.ShutDownAssociation: Shut down address: akka.trttl.tcp://remote@127.0.0.1:2553
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]
[INFO] [08/09/2016 09:25:49.120] [remote-akka.actor.default-dispatcher-15] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:49.120] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:2553/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:49.136] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:49.140] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:38701]
[INFO] [08/09/2016 09:25:49.140] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:38701]
[INFO] [08/09/2016 09:25:49.159] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38701/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:49.159] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38701/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/09/2016 09:25:49.161] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:49.161] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38701/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:25:53.175] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:53.178] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:43697]
[INFO] [08/09/2016 09:25:53.178] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:43697]
[INFO] [08/09/2016 09:25:53.195] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:43697/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:53.195] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:43697/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/09/2016 09:25:53.197] [remote-akka.actor.default-dispatcher-5] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:53.197] [remote-akka.remote.default-remote-dispatcher-7] [akka.trttl.tcp://remote@127.0.0.1:43697/system/remoting-terminator] Remoting shut down.
Expected Set(Actor[akka://ActorsLeakSpec/system/endpointManager#1423904439], Actor[akka://ActorsLeakSpec/system/transports#1152699934], Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp61#-1527258615], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp14#1344748334]), but got Set(Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp61#-1527258615], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp14#1344748334], Actor[akka://ActorsLeakSpec/system/transports#1152699934], Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A50508-9#-245584626], Actor[akka://ActorsLeakSpec/system/endpointManager#1423904439]) (ActorsLeakSpec.scala:209)[INFO] [08/09/2016 09:25:57.310] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:25:57.314] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:36743]
[INFO] [08/09/2016 09:25:57.314] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:36743]
[INFO] [08/09/2016 09:25:57.331] [remote-akka.remote.default-remote-dispatcher-4] [akka.trttl.tcp://remote@127.0.0.1:36743/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:25:57.331] [remote-akka.remote.default-remote-dispatcher-4] [akka.trttl.tcp://remote@127.0.0.1:36743/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/09/2016 09:25:57.333] [remote-akka.actor.default-dispatcher-13] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:25:57.333] [remote-akka.remote.default-remote-dispatcher-5] [akka.trttl.tcp://remote@127.0.0.1:36743/system/remoting-terminator] Remoting shut down.
[INFO] [08/09/2016 09:26:01.446] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/09/2016 09:26:01.449] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.trttl.tcp://remote@127.0.0.1:38075]
[INFO] [08/09/2016 09:26:01.449] [pool-48-thread-1-ScalaTest-running-ActorsLeakSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.trttl.tcp://remote@127.0.0.1:38075]
[INFO] [08/09/2016 09:26:01.479] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38075/system/remoting-terminator] Shutting down remote daemon.
[INFO] [08/09/2016 09:26:01.480] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38075/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [08/09/2016 09:26:03.493] [remote-akka.actor.default-dispatcher-4] [akka://remote/deadLetters] Message [akka.remote.EndpointWriter$FlushAndStop$] from Actor[akka://remote/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2FActorsLeakSpec%40127.0.0.1%3A2552-0/endpointWriter#-700939260] to Actor[akka://remote/deadLetters] was not delivered. [1] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[WARN] [08/09/2016 09:26:03.495] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Shutdown finished, but flushing might not have been successful and some messages might have been dropped. Increase akka.remote.flush-wait-on-shutdown to a larger value to avoid this.
[INFO] [08/09/2016 09:26:03.495] [remote-akka.actor.default-dispatcher-2] [akka.remote.Remoting] Remoting shut down
[INFO] [08/09/2016 09:26:03.495] [remote-akka.remote.default-remote-dispatcher-6] [akka.trttl.tcp://remote@127.0.0.1:38075/system/remoting-terminator] Remoting shut down.
[WARN] [08/09/2016 09:26:13.518] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-5] [akka.remote.Remoting] Association to [akka.trttl.tcp://remote@127.0.0.1:38075] having UID [1903183485] is irrecoverably failed. UID is now quarantined and all messages to this UID will be delivered to dead letters. Remote actorsystem must be restarted to recover from this situation.
[info] - must not leak actors (24 seconds, 810 milliseconds)
Coroner Thread Count started at 22, ended at 10, peaked at 37 in akka.remote.ActorsLeakSpec
[info] ScalaTest
[info] Run completed in 25 seconds, 157 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 1, Failed 0, Errors 0, Passed 1
[success] Total time: 26 s, completed Aug 9, 2016 9:26:14 AM
```

--- a/akka-remote/src/test/scala/akka/remote/ActorsLeakSpec.scala
+++ b/akka-remote/src/test/scala/akka/remote/ActorsLeakSpec.scala
@@ -204,10 +204,7 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
 
       EventFilter[TimeoutException](occurrences = 1).intercept {}
 
-      val finalActors = targets.flatMap(collectLiveActors).toSet
-
-      assertResult(initialActors)(finalActors)
-
+      awaitAssert(assertResult(initialActors)(targets.flatMap(collectLiveActors).toSet), 5.seconds)
     }
 
   }

